test/unit/config_test.rb

```diff
@@ -0,0 +1,420 @@
+
+require 'helper'
+require 'stringio'
+
+require 'inspec/config'
+
+describe 'Inspec::Config' do
+
+  # ========================================================================== #
+  #                                Constructor
+  # ========================================================================== #
+  describe 'the constructor' do
+    describe 'when no args are provided' do
+      it 'should initialize properly' do
+        cfg = Inspec::Config.new
+        cfg.must_respond_to :final_options
+      end
+    end
+
+    describe 'when CLI args are provided' do
+      it 'should initialize properly' do
+        cfg = Inspec::Config.new({color: true, log_level: 'warn'})
+        cfg.must_respond_to :final_options
+      end
+    end
+
+    # TODO: add test for reading from default config path
+
+  end
+
+  # ========================================================================== #
+  #                              File Validation
+  # ========================================================================== #
+  describe 'when validating a file' do
+    let(:cfg) { Inspec::Config.new({}, cfg_io) }
+    let(:cfg_io) { StringIO.new(ConfigTestHelper.fixture(fixture_name)) }
+    let(:seen_fields) { cfg.final_options.keys.sort }
+
+    describe 'when the file is a legacy file' do
+      let(:fixture_name) { 'legacy' }
+      it 'should read the file successfully' do
+        expected = ['color', 'reporter', 'target_id', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file is a valid v1.1 file' do
+      let(:fixture_name) { 'basic' }
+      it 'should read the file successfully' do
+        expected = ['create_lockfile', 'reporter', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file is minimal' do
+      let(:fixture_name) { 'minimal' }
+      it 'should read the file successfully' do
+        expected = ['reporter', 'type'].sort
+        seen_fields.must_equal expected
+      end
+    end
+
+    describe 'when the file has malformed json' do
+      let(:fixture_name) { 'malformed_json' }
+      it 'should throw an exception' do
+        ex = proc { cfg }.must_raise(Inspec::ConfigError::MalformedJson)
+        # Failed to load JSON configuration: 765: unexpected token at '{ "hot_garbage": "a", "version": "1.1",
+        # '
+        # Config was: "{ \"hot_garbage\": \"a\", \"version\": \"1.1\", \n"
+        ex.message.must_include 'Failed to load JSON config'  # The message
+        ex.message.must_include 'unexpected token'  # The specific parser error
+        ex.message.must_include 'hot_garbage' # A sample of the unacceptable contents
```

Comment:
I appreciate your example data. 

![I see you.](https://user-images.githubusercontent.com/517302/51396625-697f8380-1b0d-11e9-9c03-948a23ea174a.gif)